fluent/syntax/ast.py

```diff
@@ -61,6 +75,48 @@ def visit(value):
 
         return fun(node)
 
+    def equals(self, other, with_spans=True):
+        self_keys = set(vars(self).keys())
+        other_keys = set(vars(other).keys())
+
+        if not with_spans:
+            self_keys.discard('span')
+            other_keys.discard('span')
+
+        if self_keys != other_keys:
+            return False
+
+        keys = sorted(self_keys)
+
+        self_fields = [getattr(self, key) for key in keys]
+        other_fields = [getattr(other, key) for key in keys]
+
+        for key, field1, field2 in izip(keys, self_fields, other_fields):
```

Comment:
I'd just iterate over keys, unsorted, and assign field1 and field2 inside the loop